handlers.go

```diff
@@ -172,11 +172,33 @@ func (dht *IpfsDHT) handlePutValue(ctx context.Context, p peer.ID, pmes *pb.Mess
 	}
 	cleanRecord(rec)
 
+	// Make sure the record is valid (not expired, valid signature etc)
 	if err = dht.Validator.VerifyRecord(rec); err != nil {
 		log.Warningf("Bad dht record in PUT from: %s. %s", p.Pretty(), err)
 		return nil, err
 	}
 
+	// Make sure the new record is "better" than the record we have locally.
+	// This prevents a record with for example a lower sequence number from
+	// overwriting a record with a higher sequence number.
```

Comment:
What do you mean by this?

Reply:
The ongoing sage with keeping our old seqno's in go-ipfs :)

Reply:
Nope... still in the dark here.

Reply:
What I mean is that we have make sure we accurately track our sequence numbers, so that we can make new records with incremented seqnos.
I am referring to the issues with the disappearance of old records from the datastore in go-ipfs.
There is no bug here, just good intentions :)

Reply:
> What I mean is that we have make sure we accurately track our sequence numbers, so that we can make new records with incremented seqnos.
I am referring to the issues with the disappearance of old records from the datastore in go-ipfs.

Ah. Actually, I'm pretty sure that doesn't matter. If the last record published has expired, *any* new record published will be considered "better" than that record, regardless of the sequence number. Otherwise, we'd never be able to forget old records (our own or otherwise).

Reply:
ok, cool.

Reply:
I would like to hear from someone who designed the current republishing system, to make sure we're not breaking any assumptions, particularly their thoughts on the scenario I outlined in this comment: https://github.com/ipfs/go-ipfs/pull/4753#discussion_r172035360

Any idea who might had that knowledge @Stebalien ?

Reply:
So, I *was* able to convince @whyrusleeping that this wasn't really necessary. However, he'd still *like* it IPNS publishers remembered their last sequence number. Regardless, I believe that information belongs in the Nameservice.

Reply:
Sorry @Stebalien I don't follow.. could you explain what it is exactly that is not really necessary?

Reply:
Assuming that no record expires *before* it's predecessor expires (otherwise, bad things happen and we shouldn't be forgetting about previous records)...

**Case 1:** There exists some other *valid* (not outdated) record that we don't know about. In this case, having a copy of the outdated record doesn't help because we don't know about the latest record.

**Case 2:** There exist no other valid records that we don't know about. In this case, *any* valid record will be considered "better" than all existing records.